### ISSUE TITLE
rekey panic with `pgp-keys` defined but not set

### ISSUE THREAD

**Issue by Xianic** [hashicorp/vault#7485]
**Describe the bug**
index out of range when rekeying with zero keys

**To Reproduce**
1. Run a single instance of vault locally with a file backend
2. Run initialize the vault with `vault operator init -key-threshold=1 -key-shares=1`
3. Unseal the vault as normal
4. Run `vault operator rekey -init -key-threshold=2 -key-shares=0 -pgp-keys=`

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/hashicorp/vault/helper/pgpkeys.ReadPGPFile(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/gopath/src/github.com/hashicorp/vault/helper/pgpkeys/flag.go:100 +0x83a
github.com/hashicorp/vault/helper/pgpkeys.ParsePGPKeys(0xc00050f000, 0x1, 0x1, 0x1, 0xc00050f000, 0x1, 0x1, 0x0)
	/gopath/src/github.com/hashicorp/vault/helper/pgpkeys/flag.go:88 +0x357
github.com/hashicorp/vault/helper/pgpkeys.(*PubKeyFilesFlag).Set(0xc0005a0130, 0x7ffcbc678208, 0x0, 0x0, 0x0)
	/gopath/src/github.com/hashicorp/vault/helper/pgpkeys/flag.go:54 +0xd5
flag.(*FlagSet).parseOne(0xc0002af860, 0x1, 0x0, 0x0)
	/goroot/src/flag/flag.go:906 +0x37b
flag.(*FlagSet).Parse(0xc0002af860, 0xc0000b4030, 0x4, 0x4, 0x2effdc0, 0x2c60aa0)
	/goroot/src/flag/flag.go:925 +0x62
github.com/hashicorp/vault/command.(*FlagSets).Parse(0xc000495ec0, 0xc0000b4030, 0x4, 0x4, 0xc0000bc001, 0xc00067dcd0)
	/gopath/src/github.com/hashicorp/vault/command/base.go:432 +0x4d
github.com/hashicorp/vault/command.(*OperatorRekeyCommand).Run(0xc0005a0100, 0xc0000b4030, 0x4, 0x4, 0xc0001e2d00)
	/gopath/src/github.com/hashicorp/vault/command/operator_rekey.go:226 +0x63
github.com/hashicorp/vault/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc000279400, 0xc000279400, 0xc0000bc3c0, 0xc0000bc0c0)
	/gopath/src/github.com/hashicorp/vault/vendor/github.com/mitchellh/cli/cli.go:255 +0x207
github.com/hashicorp/vault/command.RunCustom(0xc0000b4010, 0x6, 0x6, 0xc000274410, 0x0)
	/gopath/src/github.com/hashicorp/vault/command/main.go:177 +0x5e4
github.com/hashicorp/vault/command.Run(0xc0000b4010, 0x6, 0x6, 0xc0000a0058)
	/gopath/src/github.com/hashicorp/vault/command/main.go:87 +0x48
main.main()
	/gopath/src/github.com/hashicorp/vault/main.go:10 +0x62
```

**Expected behavior**
An error explaining that `-key-shares=0` makes no sense

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.1.3
* Vault CLI Version (retrieve with `vault version`): Vault v1.1.3 ('9bc820f700f83a7c4bcab54c5323735a581b34eb')
* Server Operating System/Architecture: Ubuntu 19.04

Vault server configuration file(s):

```hcl
ui = true

backend "file" {
  path = "store"
}

listener "tcp" {
  tls_disable = 1
}
```

**Comment by jefferai** [comment]
It's actually not key-shares=0 that is the problem, it's the empty list at the end of pgp-keys.

**Comment by aivinog1** [comment]
@chrishoffman Hello, can I take this issue? I'm new in Go, but this issue is pretty clear for me.

**Comment by chrishoffman** [comment]
Fixed by #7526
